src/components/node-list/node-list-tree.js

```diff
@@ -162,9 +168,16 @@ const TreeListProvider = ({
   const renderTree = (tree, modularPipelineID) => {
     // current tree node to render
     const node = tree[modularPipelineID];
+
     if (!node) {
       return;
     }
+    const allChildrenDataEmpty = node.children.every(
+      (child) => Object.keys(child.data).length === 0
+    );
+    if (allChildrenDataEmpty) {
+      return;
+    }
```

Comment:
Could you please explain why we need `allChildrenDataEmpty` check ?

Reply:
hey good question, i'll add a comment to explain. But in short, because we need to filter out all the nodes from the node-list, that are not in slicing pipeline, and the way I filter it out is to assign an empty data object to its child. ([this change specifically](https://github.com/kedro-org/kedro-viz/blob/9d7d9cefd60364889919e8ea7c6f2fa64ccdf3ba/src/selectors/nodes.js#L291))

So the check here to make sure if the all the child under a modular pipeline are empty, then don't render that modular pipeline at all. Hope that makes sense?